biosys/apps/main/backends/auth.py

```diff
@@ -0,0 +1,42 @@
+import requests
+from django.contrib.auth import get_user_model
+
+UserModel = get_user_model()
+
+ASMS_LOGIN_URL = "https://environment.nsw.gov.au/asmslightprofileapp/account/login"
+
+
+class OEHBackend(object):
+    """
+    Connect to the OEH ASMS login page and check the user credentials.
+    """
+    user_model = get_user_model()
+
+    @staticmethod
+    def asms_login_valid(username, password):
+        payload = {
+            'UserName': username,
+            'Password': password
+        }
+        response = requests.post(ASMS_LOGIN_URL, data=payload)
+        # to test for success look for a redirection.
+        has_redirection = False
+        if response.history and response.history[0].status_code in [301, 302]:
+            has_redirection = True
+        return response.status_code == 200 and has_redirection
+
+    def authenticate(self, request, username=None, password=None):
+        asms_valid = self.asms_login_valid(username, password)
+        user = None
+        if asms_valid:
+            user, _ = UserModel.objects.get_or_create(username=username)
+            # update password every time.
+            user.set_password(password)
```

Comment:
what about some functionality here to get the user's first name, last name and email?

Reply:
I didn't want that for two reasons:
* It will add another delay to the login. The asms login can be a bit slow.
* I didn't want to bring the all beautifulsoup html parsing unless absolutely necessary.
I'm waiting for Bill to tell me if there's a better way to get user info.
It is not absolutely necessary at this point.

Reply:
That's four reasons.

Reply:
Two bullet points

Reply:
Just because there's only two bullet points, doesn't mean there's only two reasons.

I'll stop now.